### PR TITLE
VAULT-23122: Audit fix for 'log_raw' issue

### PR DESCRIPTION
**Pull request by peteski22**
This PR fixes a bug where the use of `log_raw` on any audit devices may have caused other devices (not configured with `log_raw`) to log raw audit data.

Please see: https://developer.hashicorp.com/vault/docs/upgrading/upgrade-to-1.15.x#audit-devices-could-log-raw-data-despite-configuration

Related PRs which also partially improve/fix this issue:

* https://github.com/hashicorp/vault/pull/24811
* https://github.com/hashicorp/vault/pull/24843
* https://github.com/hashicorp/vault/pull/24885

### CODE REVIEW

File: audit/entry_formatter.go [hunk 1]
```diff
@@ -153,10 +153,25 @@ func (f *EntryFormatter) Process(ctx context.Context, e *eventlogger.Event) (*ev
...
+		Version:   a.Version,
+		Subtype:   a.Subtype,
+		Timestamp: a.Timestamp,
+		Data:      data, // Use the cloned data here rather than a pointer to the original.
+	}
+
+	e2 := &eventlogger.Event{
+		Type:      e.Type,
+		CreatedAt: e.CreatedAt,
+		Formatted: e.Formatted,
```
Comment (ncabatoff) [inline audit/entry_formatter.go:168]: As maps are pointers, you may wish to do something different here.

File: internal/observability/event/sink_socket.go
Comment (peteski22) [inline internal/observability/event/sink_socket.go:1]: formatting tweaks

File: internal/observability/event/sink_stdout.go
Comment (peteski22) [inline internal/observability/event/sink_stdout.go:1]: formatting tweaks (we use `e` in other `Process` funcs)

File: internal/observability/event/sink_syslog.go
Comment (peteski22) [inline internal/observability/event/sink_syslog.go:1]: formatting tweaks

**Comment by marcboudreau** [comment]
I will take some time over the weekend to better understand this issue to make sure that we are fixing it completely.

**Review by kubawi — APPROVED** [review]

**Pull request merged**